src/pow.rs

```diff
@@ -8,21 +8,59 @@ use std::{
 };
 
 use arrayvec::ArrayVec;
+use bdk_wallet::bitcoin::Amount;
 use concurrent_map::{CasFailure, ConcurrentMap};
 use parking_lot::{Mutex, MutexGuard};
 use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
 use sha2::{Digest, Sha256};
 use terrors::OneOf;
 use tokio::time::sleep;
 
-use crate::err;
+use crate::{err, settings::SETTINGS};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Challenge {
     nonce: Nonce,
+    claimed: bool,
     expires_at: Instant,
+    difficulty: u8,
 }
 
-const TTL: Duration = Duration::from_secs(60);
+#[derive(Deserialize, Serialize, Debug)]
+pub struct PowConfig {
+    /// Minimum balance required for a user to claim funds. Defaults to 500 BTC,
+    /// or 50_000_000_000 sats. When configuring in the config file, this value
+    /// should be in sats as a number.
```

Comment:
```suggestion
    /// Minimum balance required for a user to claim funds.
    ///
    /// Defaults to `500` BTC, or `50_000_000_000` sats.
    /// When configuring in the config file, this value
    /// should be in sats as a number.
```

What number means?
Can it be `2.0`, `3`, `2+5i`, `0xfe`.

All of these are valid "numbers".